mappers/modbus/device/device.go

```diff
@@ -0,0 +1,251 @@
+		return
+	}
+	klog.Info("Device id: ", id)
+
+	var device *globals.ModbusDev
+	var ok bool
+	if device, ok = devices[id]; !ok {
+		klog.Error("Device not exist")
+		return
+	}
+
+	// Get twin map key as the propertyName
+	var delta mappercommon.DeviceTwinDelta
+	if err := json.Unmarshal(message.Payload(), &delta); err != nil {
+		klog.Error("Unmarshal message failed: ", err)
+		return
+	}
+	for twinName, twinValue := range delta.Delta {
+		i := 0
+		for i = 0; i < len(device.Instance.Twins); i++ {
+			if twinName == device.Instance.Twins[i].PropertyName {
+				break
+			}
+		}
+		if i == len(device.Instance.Twins) {
+			klog.Error("Twin not found: ", twinName)
+			continue
+		}
+		// Type transfer
+		device.Instance.Twins[i].Desired.Value = twinValue
+		var visitorConfig configmap.ModbusVisitorConfig
+		if err := json.Unmarshal([]byte(device.Instance.Twins[i].PVisitor.VisitorConfig), &visitorConfig); err != nil {
+			klog.Error("Unmarshal visitor config failed")
+		}
+		valueConverted, err := mappercommon.Convert(device.Instance.Twins[i].PVisitor.PProperty.DataType, twinValue)
+		if err == nil {
+			valueInt, _ := valueConverted.(int64)
+			device.ModbusClient.Set(visitorConfig.Register, visitorConfig.Offset, uint16(valueInt))
+		} else {
+			klog.Error("Convert failed")
+		}
+	}
+}
+
+// isRS485Enabled is RS485 feature enabled for RTU.
+func isRS485Enabled(customizedValue configmap.CustomizedValue) bool {
+	isEnabled := false
+
+	if len(customizedValue) != 0 {
+		if value, ok := customizedValue["serialType"]; ok {
+			if value == "RS485" {
+				isEnabled = true
+			}
+		}
+	}
+	return isEnabled
+}
+
+// initModbus initialize modbus client
+func initModbus(protocolConfig configmap.ModbusProtocolCommonConfig, slaveID int16, client *driver.ModbusClient) error {
+	if protocolConfig.Com.SerialPort != "" {
+		modbusRtu := driver.ModbusRtu{SlaveId: byte(slaveID),
+			SerialName:   protocolConfig.Com.SerialPort,
+			BaudRate:     int(protocolConfig.Com.BaudRate),
+			DataBits:     int(protocolConfig.Com.DataBits),
+			StopBits:     int(protocolConfig.Com.StopBits),
+			Parity:       protocolConfig.Com.Parity,
+			RS485Enabled: isRS485Enabled(protocolConfig.CustomizedValues),
+			Timeout:      5 * time.Second}
+		*client, _ = driver.NewClient(modbusRtu)
+	} else if protocolConfig.Tcp.Ip != "" {
+		modbusTcp := driver.ModbusTcp{
+			SlaveId:  byte(slaveID),
+			DeviceIp: protocolConfig.Tcp.Ip,
+			TcpPort:  strconv.FormatInt(protocolConfig.Tcp.Port, 10),
+			Timeout:  5 * time.Second}
+		*client, _ = driver.NewClient(modbusTcp)
+	} else {
+		return errors.New("No protocol found")
+	}
+	return nil
+}
+
+// initTwin initialize the timer to get twin value.
+func initTwin(dev *globals.ModbusDev) {
+	for i := 0; i < len(dev.Instance.Twins); i++ {
+		var visitorConfig configmap.ModbusVisitorConfig
+		if err := json.Unmarshal([]byte(dev.Instance.Twins[i].PVisitor.VisitorConfig), &visitorConfig); err != nil {
+			klog.Error(err)
+			continue
+		}
```

Comment:
well, I think just ReadWrite AccessMode needs to convert because has Desired value but not for ReadOnly AccessMode
```
md.Instance.Twins[i].PVisitor.PProperty.AccessMode == AssessModeRW{
		value, err := mappercommon.Convert(dev.Instance.Twins[i].PVisitor.PProperty.DataType, dev.Instance.Twins[i].Desired.Value)
		if err != nil {
			klog.Error("Convert error. Type: %s, value: %s ", dev.Instance.Twins[i].PVisitor.PProperty.DataType, dev.Instance.Twins[i].Desired.Value)
			continue
		}
		valueInt, _ := value.(int64)
		_, err = dev.ModbusClient.Set(visitorConfig.Register, visitorConfig.Offset, uint16(valueInt))
		if err != nil {
			klog.Error(err, visitorConfig)
			continue
		}
}
```

Reply:
Done. Add this check.